sycl/include/sycl/vector.hpp

```diff
@@ -1315,3 +1317,5 @@ class SwizzleOp {
 } // namespace detail
 } // namespace _V1
 } // namespace sycl
+
+#include <sycl/detail/vector_convert.hpp>
```

Comment:
The file is included this way, because it has dependency on both `rounding_mode` and `vec` (we can't define `vec::convert` until we have a declaration for it).

An alternative is to have `vector_convert` to be included separately by every user of `vec::convert` method, but that doesn't seem like a good idea. Perhaps some other tricks are possible, which are better, feedback is appreciated here.

Reply:
This change was reverted in b38ceed5718bbecb7ef866cba66e4651f6ef560f